app/ui-react/syndesis/src/modules/integrations/components/editor/utils.ts

```diff
@@ -43,6 +45,30 @@ type StepKindHrefCallback = (
   s: ISelectConnectionRouteState
 ) => H.LocationDescriptorObject;
 
+/**
+ * Checks whether or not the provided object is a technical preview.
+ * Accepts either an IUIStep, Connection, or a Connector.
+ * Returns a boolean for whether or not the metadata `tech-preview` key
+ * returns a string value of 'true'
+ * @param data
+ */
+export function isTechPreview<T extends IUIStep | Connection | Connector>(
```

Comment:
Do we really need to accept all these types? `Step`s don't have a `connection`, and neither `Extension`s do.

We really just want to accept `Connector` here, since the tech preview metadata is stored there.

Then [this](https://github.com/syndesisio/syndesis-react/pull/409/files#diff-c8c891bfe81f60cb81839bf4031cb8c7R10) can consume this function and set the `isTechPreview` flag.

Bonus points for:
* enhancing the `useConnector` to provide a transformation function (like the `useConnection` does)
* change the `WithConnector` and `WithConnectors` component to use the connector transformation function
* enhance the `transformConnectionResponse` to run the new transformation function on the connection's connector, and retrieve the `isTechPreview` flag from the transformed connector object.

Reply:
>Do we really need to accept all these types? Steps don't have a connection, and neither Extensions do.

That was in reference to this feedback: https://github.com/syndesisio/syndesis-react/pull/409#discussion_r291590677
I'll change it to `Connector` 🤷‍♀

Reply:
Yes, that was a step toward a better refactoring. My suggestion aims to not having to rely on those helpers across the app and enhance the model so that the information is already available for the consumer at all times. If you do what I suggested you can enhance the `IUIStep` interface to also have `isTechPreview` and `isConfigRequired` and you can set those properties to false for steps and extensions [here](https://github.com/syndesisio/syndesis-react/blob/02625588f49dd054095fa8a19d809c099e1aa71f/app/ui-react/syndesis/src/modules/integrations/components/editor/utils.ts#L57) where the `IUIStep` object is generated.

Reply:
I'm following but am not sure if I'm doing this properly, please see my changes. Also, was easier to start with `WithConnectors.tsx` first.